libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/Store.kt

```diff
@@ -68,20 +68,17 @@ open class Store {
     }
 
     fun dispatch(action: Action) {
-        if (isDispatching) {
-            logW(
-                "Cannot dispatch new actions during reducer processing. " +
-                    "Action dropped: ${action::class.java.simpleName}.",
-                "Store"
-            )
-            return
+        actions.add(action)
+        if (isProcessing) return
+
+        isProcessing = true
+        while (actions.isNotEmpty()) {
```

Comment:
this is happening on the main thread correct 😱

Reply:
This should be in a coroutine scope at least, to make it so the main thread can be free'd up every once and awhile.

This thing would lock the main thread if there are continuous actions

Reply:
Or it would be good to have a "dispatcher" that is responsible for watching the actions

Reply:
Having continuous actions looks like an error to me. But even if we process actions on a background thread, most of the state collectors, if not all of them, run on the main thread, so performance will be affected anyway.

Reply:
I'm not suggesting a background thread. Considered a coroutine context switch in order to "free up" the main thread in between actions. Essentially what `MainScope()` does.

https://android.googlesource.com/platform/frameworks/support/+/a9ac247af2afd4115c3eb6d16c05bc92737d6305/lifecycle/runtime/src/main/java/androidx/lifecycle/LifecycleRegistry.java#138

On the other hand the `LifecycleRegistry` does the same thing you're proposing here. It "syncs" all observers and this is done on the main thread while blocking.

Reply:
Could explore dispatching it later, resolving this thread

Reply:
actually thinking about it. the `actions` could become stale. I mean, actions will be added to the list and not dispatched. it's only when a new action is called that previous ones will be dispatched